### ISSUE TITLE
problems with libm detection for ceil()

### ISSUE THREAD

**Issue by sthen** [net-snmp/net-snmp#72]
7497e2eade tries to detect if libm is required for ceil() (as used in snmplib/keytools.c if configure is called with --enable-blumenthal-aes), but there are some problems.

- An earlier check for exp() - "diskIOLAx requires exp which may require -lm" in configure.d/config_os_libs1 leaves -lm in LIBS which is added to other checks including the one for ceil, so if exp requires -lm then you always get "checking for ceil without -lm... yes".

- The ceil() check used is a link check for code calling "ceil((double)2.4)". Some compilers (at least I see it on clang 8.0.1 on OpenBSD) optimize away ceil with a fixed constant so you also always get "checking for ceil without -lm... yes" with such a compiler.

- Even when it's detected as needed for ceil(), -lm is only added to LIBS not LNETSNMPLIBS so there's an unresolved symbol in the library. (Tools do pull in libm so work ok, but other software linking the library fails).

In my case -lm is already added to LIBS as needed from the abs() check and I can add the library linkage with ```AC_CHECK_LIB(m, ceil, [LNETSNMPLIBS="$LNETSNMPLIBS -lm"])``` in configure.d/config_os_libs2 but it probably needs something smarter in the more general case.

**Comment by bvanassche** [comment]
Please retest with the latest version of the master branch.

**Comment by sthen** [comment]
Thanks Bart, it mostly helps but -lm is still present in LIBS from the exp() test in config_os_libs1.

```
configure:25205: checking for library containing ceil
configure:25233: cc -o conftest -DNETSNMP_ENABLE_IPV6 -fno-strict-aliasing -DNETSNMP_REMOVE_U64 -O2 -pipe -g -Uopenbsd6 -Dopenbsd6=openbsd6   conftest.c   -lm  >&5
conftest.c:233:6: warning: incompatible redeclaration of library function 'ceil' [-Wincompatible-library-redeclaration]
char ceil ();
     ^
conftest.c:233:6: note: 'ceil' is a builtin with type 'double (double)'
1 warning generated.
configure:25233: $? = 0
configure:25269: result: none required
```

I guess the simplest fix is to save/clear libs when running the ceil test? With this change it works for me:

```
+netsnmp_save_LIBS="$LIBS"
+LIBS=
 NETSNMP_SEARCH_LIBS(ceil, m,,,, LNETSNMPLIBS)
+LIBS="$netsnmp_save_LIBS"
```

**Comment by bvanassche** [comment]
An additional fix has been checked in on the v5.8 and master branches. Please retest.

**Comment by sthen** [comment]
Many thanks, that works nicely here.
